impeller/typographer/backends/skia/typographer_context_skia.cc

```diff
@@ -376,141 +384,127 @@ std::shared_ptr<GlyphAtlas> TypographerContextSkia::CreateGlyphAtlas(
-        break;
-      case GlyphAtlas::Type::kColorBitmap:
-        descriptor.format = PixelFormat::kR8G8B8A8UNormInt;
-        break;
-    }
-    descriptor.size = atlas_size;
-    descriptor.storage_mode = StorageMode::kDevicePrivate;
-    descriptor.usage = TextureUsage::kShaderRead | TextureUsage::kRenderTarget;
-    new_texture = context.GetResourceAllocator()->CreateTexture(descriptor);
+  TextureDescriptor descriptor;
+  switch (type) {
+    case GlyphAtlas::Type::kAlphaBitmap:
+      descriptor.format =
+          context.GetCapabilities()->GetDefaultGlyphAtlasFormat();
+      break;
+    case GlyphAtlas::Type::kColorBitmap:
+      descriptor.format = PixelFormat::kR8G8B8A8UNormInt;
+      break;
   }
-
+  descriptor.size = atlas_size;
+  descriptor.storage_mode = StorageMode::kDevicePrivate;
+  descriptor.usage = TextureUsage::kShaderRead;
+  std::shared_ptr<Texture> new_texture =
+      context.GetResourceAllocator()->CreateTexture(descriptor);
   if (!new_texture) {
     return nullptr;
   }
 
   new_texture->SetLabel("GlyphAtlas");
 
   std::shared_ptr<CommandBuffer> cmd_buffer = context.CreateCommandBuffer();
-  std::shared_ptr<BlitPass> blit_pass;
+  std::shared_ptr<BlitPass> blit_pass = cmd_buffer->CreateBlitPass();
 
   // The R8/A8 textures used for certain glyphs is not supported as color
-  // attachments in most graphics drivers. To be safe, just do a CPU clear
-  // for these.
-  if (type == GlyphAtlas::Type::kAlphaBitmap) {
-    size_t byte_size =
-        new_texture->GetTextureDescriptor().GetByteSizeOfBaseMipLevel();
-    BufferView buffer_view =
-        host_buffer.Emplace(nullptr, byte_size, DefaultUniformAlignment());
-
-    ::memset(buffer_view.buffer->OnGetContents() + buffer_view.range.offset, 0,
-             byte_size);
-    buffer_view.buffer->Flush();
-    blit_pass = cmd_buffer->CreateBlitPass();
-    blit_pass->AddCopy(buffer_view, new_texture);
-  } else {
-    // In all other cases, we can use a render pass to clear to a transparent
-    // color.
-    ColorAttachment attachment;
-    attachment.clear_color = Color::BlackTransparent();
-    attachment.load_action = LoadAction::kClear;
-    attachment.store_action = StoreAction::kStore;
-    attachment.texture = new_texture;
-
-    RenderTarget render_target;
-    render_target.SetColorAttachment(attachment, 0u);
-
-    auto render_pass = cmd_buffer->CreateRenderPass(render_target);
-    render_pass->EncodeCommands();
-    blit_pass = cmd_buffer->CreateBlitPass();
-  }
-  FML_DCHECK(!!blit_pass);
+  // attachments in most graphics drivers. For other textures, most framebuffer
+  // attachments have a much smaller size limit than the max texture size.
+  size_t byte_size =
+      new_texture->GetTextureDescriptor().GetByteSizeOfBaseMipLevel();
+  BufferView buffer_view =
+      host_buffer.Emplace(nullptr, byte_size, DefaultUniformAlignment());
+
+  ::memset(buffer_view.buffer->OnGetContents() + buffer_view.range.offset, 0,
```

Comment:
Unclear about why this is necessary as the blit will clear the contents and we shouldn't be sampling from locations that aren't cleared.

Reply:
Assuming this is necessary, let's add a trace around this?

Reply:
Done